### PR TITLE
maintainer: add a job with the minimum required boost version

### LINKED ISSUE #2837: Utils function not compatible with required boost version
```Utils::make_lin_space``` does not build with boost version 1.53 which is currently the version we require. By inspection of the boost repo, I suppose we need at least 1.57 (from this version on at least the namespaces exist).

### PR DESCRIPTION
**Pull request by KaiSzuttor**
Fixes #2837

### CODE REVIEW

**Comment by KaiSzuttor** [comment]
this needs to be merged before we can test the new container in the docker PR. After that we can activate the job in the CI.

**Comment by KaiSzuttor** [comment]
@fweik any idea, what is happening there?
https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/116554

**Comment by mkuron** [comment]
> https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/116554
> mpiexec noticed that process rank 1 with PID 0 on node coyote exited on signal 4 (Illegal instruction).

That also happened last night in the i386 container: https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/116520 (#2851). Sounds like the compiler is generating instructions not supported by the processor?

**Comment by KaiSzuttor** [comment]
@jngrad I don't understand the failing samples test. If I locally run the sample ```observables_correlators.py``` I get the exact same error for the current python branch. However, the CI did only detect the issue in this PR. Could you please have a look into that?

**Comment by jngrad** [comment]
@KaiSzuttor found it: `Accumulators::Correlator::set_correlation_args()` in the core is never called by the interface constructor `ScriptInterface::Accumulators::Correlator::Correlator()`, so the value `[10, 10, 10]` passed to argument `args` are not copied to `m_correlation_args`, which is left uninitialized, usually with values `[0., 0., 0.]`, so no throw (and returns wrong numerical results).
https://github.com/espressomd/espresso/blob/0520af905d8914c284b672e9c0493414a063c66b/src/script_interface/accumulators/Correlator.hpp#L43-L53

**Comment by KaiSzuttor** [comment]
But why is this not an issue in the current python branch?

File: src/core/io/writer/h5md_core.hpp [hunk 1]
```diff
@@ -27,17 +27,15 @@
 #include "cells.hpp"
 #include "global.hpp"
 
+#include "communication.hpp"
```
Comment (fweik) [inline src/core/io/writer/h5md_core.hpp]: Please clean up the includes. I think the core headers are only needed in the implementation..

**Review by fweik — CHANGES_REQUESTED** [review]

**Comment by jngrad** [comment]
The scenario `corr_operation="fcs_acf"` is untested, so only jobs with sample checks enabled will run it via the sample `observables_correlators.py`. When printing the member `correlation_args`, I get either `0.000000 0.000000 0.000000` or `0.000000 0.000000 -7732675196468477331125564344973318597912019529669651378575435354452340506624.000000` followed by our RuntimeError.  `T m_data[N];` (the internal data structure for `Vector3d correlation_args`) should be implicitly zero-initialized, so there must be an issue with pointers. From my side the RuntimeError frequency is rather low and very sensitive to how many print/printf functions I add in the core class or in the sample script.

**Review by fweik — DISMISSED** [review]

**Comment by fweik** [comment]
The correlator code is unreadable and incomprehensible, I don't think it is worth while investigating any issues in it.

**Comment by fweik** [comment]
@jngrad you wrote
> Accumulators::Correlator::set_correlation_args() in the core is never called by the interface constructor ScriptInterface::Accumulators::Correlator::Correlator()

The constructor of `ScriptInterface::Accumulators::Correlator::Correlator()` is not where the contained object is initialized, but by the call of `make_shared` in `ScriptInterface::Accumulators::Correlator::construct`...

**Comment by jngrad** [comment]
Yes I realized it after posting my message. There is no `args` argument in the core constructor so the call to the core initializer method checks an uninitialized core member. Providing a default argument still doesn't fix the RuntimeError though.

**Comment by jngrad** [comment]
I may have found a solution, let me check this on a clean source.

File: src/core/accumulators/Correlator.cpp [hunk 2]
```diff
@@ -298,7 +298,7 @@ void Correlator::initialize() {
     m_correlation_args[0] = m_correlation_args[0] * m_correlation_args[0];
     m_correlation_args[1] = m_correlation_args[1] * m_correlation_args[1];
     m_correlation_args[2] = m_correlation_args[2] * m_correlation_args[2];
-    fprintf(stderr, "args2: %f %f %f\n", m_correlation_args[0],
+    fprintf(stderr, "args^2: %f %f %f\n", m_correlation_args[0],
```
Comment (fweik) [inline src/core/accumulators/Correlator.cpp]: This should probably be removed.

**Review by fweik — CHANGES_REQUESTED** [review]

File: src/core/accumulators/Correlator.cpp (hunk 2 above)
Comment (jngrad) [reply src/core/accumulators/Correlator.cpp]: Removed.

**Review by fweik — APPROVED** [review]

**Comment by jngrad** [comment]
`coulomb_cloud_wall` is failing repeatedly

**Comment by KaiSzuttor** [comment]
seems to be some UB?

**Comment by fweik** [comment]
No, it's still `Illegal instruction`, so most likely some 64-bit code got in there

**Comment by KaiSzuttor** [comment]
the printed error is not necessarily the actual if you google around...

**Comment by KaiSzuttor** [comment]
https://stackoverflow.com/questions/10354147/find-which-assembly-instruction-caused-an-illegal-instruction-error-without-debu

**Comment by jngrad** [comment]
bors r=fweik

**Pull request merged**